src/components/PatientForm.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { FunctionComponent, useState, useContext, useCallback } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { Patient } from '../types'
+import { TokenContext } from '../context/token'
+import { ErrorContext, ErrorType } from '../context/error'
+
+import { strict as assert } from 'assert'
+
+import Button from 'react-bootstrap/Button'
+import Row from 'react-bootstrap/Row'
+import { updatePatient } from '../services/api'
+
+interface PatientFormProps {
+  patient: Patient
+}
+
+interface Field {
+  value: any
+  error?: string
+  isValid: boolean
+}
+
+interface Form {
+  lastname: Field
+  firstname: Field
+  address: Field
+  city: Field
+  zipcode: Field
+
+}
```

Comment:
Can't we simply use the Patient already defined in types?
The only thing I can think of that could be an issue is it's expecting an ID and I don't know how that would play when using the `updatePatient()` function